### PR TITLE
BREAKING CHANGE: Use only Vault for storage

### PR DESCRIPTION
**Pull request by aearly**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,7 +41,12 @@ Vault will have:
 - a map of Google service account keys at `/credentials/google/${serviceAccountEmail}`
 - a map of AWS IAM keys at `/credentials/amazon/${accountName}`
 - common provider specific data shared amongst clusters at `/clusters/common/${provider}`
-- a map of clusters `/clusters/${environment}/${clusterName}` containing all sensitive information specific to a single cluster
+- a `/clusters/all` record whose value is a map of all `clusterSlug`s to their path in Vault, eg `mycluster: '/clusters/production/mycluster'`
+- a map of clusters `/clusters/${environment}/${clusterSlug}` containing all information specific to a single cluster.  The record will have
+  + A `value` key with a JSON blob of general cluster info, as defined by the schema.
+  + A `channels` key with a JSON array of all channels the cluster belongs to
+- a `/channels/all` record whose `value` is a comma-separated list of all channels
```
Comment (nlf) [inline README.md]: should this be a JSON array for consistency?

Comment (aearly) [reply README.md]: Yep, that was easier to do internally as well.

File: test/index.test.js [hunk 2]
```diff
@@ -43,331 +37,442 @@ tap.test('create and close a client w/ defaults', async t => {
...
-    t.same((await redis.smembers('channels')).sort(), ['default', 'other'])
+    t.test('add another', async t => {
+      const vaultMock = nock('http://vault.dev:8200/')
+        .get('/v1/kv/data/channels/all').reply(200, kvGet({ value: '["default","dummy"]' }))
+        .put('/v1/kv/data/channels/all', kvPut({ value: ['default', 'dummy', 'other'] })).reply(200)
+        .put('/v1/kv/data/channels/other', { data: { value: '[]' } }).reply(200)
 
-    await client.createChannel('default')
-    t.same((await redis.smembers('channels')).sort(), ['default', 'other'])
+      console.log('add other')
```
Comment (nlf) [inline test/index.test.js]: this line can go away

File: test/index.test.js [hunk 3]
```diff
@@ -43,331 +37,442 @@ tap.test('create and close a client w/ defaults', async t => {
...
+      await client.createChannel('other')
+      vaultMock.done()
+    })
+
+    t.test('add existing', async t => {
+      const vaultMock = nock('http://vault.dev:8200/')
+        .get('/v1/kv/data/channels/all')
+        .reply(200, kvGet({ value: '["default","dummy","other"]' }))
+
+      console.log('add default')
```
Comment (nlf) [inline test/index.test.js]: this one too

File: test/index.test.js [hunk 4]
```diff
@@ -43,331 +37,442 @@ tap.test('create and close a client w/ defaults', async t => {
...
   t.test('createChannel', async t => {
-    await client.createChannel('default')
-    t.same(await redis.smembers('channels'), ['default'])
+    t.test('add one', async t => {
+      const vaultMock = nock('http://vault.dev:8200/')
+        .get('/v1/kv/data/channels/all').reply(200, kvGet({ value: '["dummy"]' }))
+        .put('/v1/kv/data/channels/all', kvPut({ value: ['default', 'dummy'] })).reply(200)
+        .put('/v1/kv/data/channels/default', kvPut({ value: '[]' })).reply(200)
+
+      console.log('add default')
```
Comment (nlf) [inline test/index.test.js]: almost missed it, but this line can also go away

**Review by nlf — APPROVED** [review]
i feel like it's telling that my only line comments are to remove `console.log()` calls in tests :joy:

this is _beautiful_

**Pull request merged**